### PR TITLE
wip: Makefile setup

### PR DESCRIPTION
**Pull request by t1m0t**
Added configuration files to perform `cargo make tests` so that it will run the tests in all packages where the root Makefile has a link with and allow runtime tests inside a browser.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,4 +4,5 @@
     "desktop",
     "router"
   ],
+  "editor.formatOnSave": false
```
Comment (jkelleyrtp) [inline .vscode/settings.json:7]: https://stackoverflow.com/questions/44831313/how-to-exclude-file-extensions-and-languages-from-format-on-save-in-vscode

let's do it just for the toml files

Comment (t1m0t) [reply .vscode/settings.json]: corrected

**Comment by t1m0t** [comment]
wip for #185 and #186

**Review by jkelleyrtp — CHANGES_REQUESTED** [review]
Everything looks good! But, what is the docker submodule doing here? Do we need it?

It doesn't seem to be used anywhere.

**Comment by t1m0t** [comment]
> Everything looks good! But, what is the docker submodule doing here? Do we need it?
> 
> It doesn't seem to be used anywhere.

Yes for local testing before pushing. It is useful to execute tests locally on someone's machine, so that we are sure no errors would happen in git workflows. It saves us a lot of push in the remote repo that wouldn't work. It is not mandatory but it can be useful for anyone who wants to tests his changes before pushing it.

**Comment by t1m0t** [comment]
> Everything looks good! But, what is the docker submodule doing here? Do we need it?
> 
> It doesn't seem to be used anywhere.

fixed it

**Review by jkelleyrtp — APPROVED** [review]
Thanks!!

**Pull request merged**